tests/adapter/dbt/tests/adapter/materialized_views/test_materialized_views.py

```diff
@@ -4,6 +4,7 @@
 
 from dbt.tests.util import run_dbt, get_manifest
 from dbt.contracts.relation import RelationType
+from dbt.contracts.results import RunStatus
 
 
 class MaterializedViewTestsBase:
```

Comment:
Would we expect all of these to be passable for the `skip` and `fail` option or do we need to rename these to stop test call for automatic run and just call them as we would expect in the inherited class?

Reply:
I need to think through the scenarios, but I suspect that some may not run appropriately until we are done with the feature. Either way, I would not rename the test. I would instead use the `@pytest.mark.skip()` decorator.

Reply:
will look into how flexible we can be with the skip, see if we can possibly use the configuration option to control if it skips or not

Reply:
We need to test the matrix of `['apply', 'skip', 'fail']` X `[--full-refresh, not --full-refresh]` X `[configuration_change_1, ..., configuration_change_n]`. The third one is the tricky one because it changes by database platform. I think we wind up with a base class, or set of classes, that take care of the first two dimensions, and then the adapter maintainer needs to supply the last dimension as part of implementing materialized views.

We should only expose the scenarios where the configuration change matters. For example, the `on_configuration_change` setting shouldn't matter if --full-refresh is specified, even if there are configuration changes and 'fail' is selected. In that scenario, we know the outcome, regardless of what the adapter maintainer supplies.

We should also test the scenario where there are multiple configuration changes, one of which can be applied via an alter statement, but one of which requires a full refresh. In this case, we should show that we only do a full refresh. This would be a specific test within the 'apply' scenario.

I'm not sure this entirely answers your question. We can pair up on it if you'd like.